tests/panic_source_locations.rs

```diff
@@ -0,0 +1,39 @@
+#[allow(unused_imports)]
+use cradle::prelude::*;
+#[allow(unused_imports)]
+use pretty_assertions::assert_eq;
+use std::{
+    panic::{set_hook, take_hook},
+    sync::{Arc, Mutex},
+};
+
+#[rustversion::since(1.46)]
+#[test]
+fn panics_contain_source_locations_of_run_and_run_output_call() {
+    let f = || run!("false");
+    let panic_location = get_panic_location(f);
+    assert_eq!(
+        Some("tests/panic_source_locations.rs:13:16".to_string()),
+        panic_location
+    );
+    let f = || run_output!("false");
+    let panic_location = get_panic_location(f);
+    assert_eq!(
+        Some("tests/panic_source_locations.rs:19:16".to_string()),
+        panic_location
+    );
```

Comment:
These two assertions are in the same test to make sure they're not executed in parallel. `get_panic_location` is not thread-safe, since it modifies the global panic handler. There's other ways to solve this, not sure they're better though.

Reply:
Can't this also race with other tests, if those tests panic?